spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/BravePropagationConfigurations.java

```diff
@@ -101,8 +101,18 @@ BaggagePropagation.FactoryBuilder propagationFactoryBuilder(
 		private Factory createThrowAwayFactory() {
 			return new Factory() {
 
+				/**
+				 * @deprecated end users and instrumentation should never call this, and instead
+				 * use {@link #get()}. This only remains to avoid rev-lock upgrading to Brave 6.
+				 */
+				@Deprecated
 				@Override
-				public <K> Propagation<K> create(KeyFactory<K> keyFactory) {
+				public <K> Propagation<K> create(Propagation.KeyFactory<K> ignored) {
+					throw new UnsupportedOperationException("As of Brave 5.12, call PropagationFactory.get()");
+				}
+
+				@Override
+				public Propagation<String> get() {
```

Comment:
My reading of https://github.com/openzipkin/brave/pull/1396 is that Brave 5.18 will require `get` to be implemented. Have I missed something? This change appears to be unrelated to upgrading to Brave 5.17.1.

I'm also not sure about throwing an `UnsupportedOperationException`. Is it not pushing a guaranteed failure to runtime when remove the method would have caught it at compile time instead? If the goal is to provide a smoother migration path, I wonder if `create` could delegate somehow to `get` instead.

Reply:
The issue is that `create` was formerly abstract and hasn't been called directly in a very long time.

In any case, I removed the special casing. I updated this to Brave 6, but it will also work without change with 5.18.

Reply:
since otel is already up with Brave 6 and Zipkin Reporter 3, let's not worry too much about this part. Feel free to unresolve if you disagree!